### DISCUSSION TITLE
Add support for emoji in chat messages

### DISCUSSION THREAD

**Discussion by jai-x** [ppy/osu#15182] [Ideas]
Issue pointed out by a friend who uses emoji in messages frequently.
May just be a case of adding the appropriate font files such that emoji usage is just automatically rendered by SpriteText.
Twemoji is a popular choice for free emoji sets: https://twemoji.twitter.com/

**Comment by bdach** [comment]
Not entirely related but that reminds me that we also need to update the flags with the twemoji ones in line with web. I have a half-finished bash script lying somewhere that exports twemoji svg to png. May dig it up and post here.

**Reply by peppy** [reply]
Of relevance, such a script should probably be added in osu-web, and run once to generate static assets that are published for [wiki use](https://github.com/ppy/osu-wiki/issues/6017).

**Reply by bdach** [reply]
Script for client is [here](https://gist.github.com/bdach/3bda4213ec011b37407a1f84f2a58f32).

I'm not PRing the update yet because annoyingly twemoji have different flag aspect ratio and it's messing with designs. Need to sit down properly and go over all usages rather than PRing something half arsed.
